File: android/src/main/java/com/gachlab/geolocation/domain/GeofenceTransition.kt
// SPDX-License-Identifier: MIT
// Copyright (c) 2026 gachlab

package com.gachlab.geolocation.domain

/**
 * A geofence boundary transition — the domain modelling of a geofence event
 * (ARCHITECTURE.md domain set's "GeoEvent"), the Android twin of the web/iOS
 * `GeofenceTransition`.
 *
 * Pure — no GMS import. The GMS `Geofence.GEOFENCE_TRANSITION_*` ints are mapped to this
 * enum at the platform boundary (`GeofenceBroadcastReceiver`), keeping the domain free of
 * Play-services types. The hub bus (`ServiceEvent.Geofence{Enter,Exit,Dwell}`) already
 * carried the transition; this names it in the shared vocabulary so the three dies agree.
 */
enum class GeofenceTransition { ENTER, EXIT, DWELL }

File: src/web/domain/geofence-transition.ts
// SPDX-License-Identifier: MIT
// Copyright (c) 2026 gachlab

/**
 * A geofence boundary transition — the domain modelling of a geofence event
 * (ARCHITECTURE.md domain set's "GeoEvent"), the web twin of the Android/iOS
 * `GeofenceTransition`.
 *
 * Each platform's event bus already carries the transition alongside the region id and
 * the platform location; what was loose was the *transition itself* — a magic
 * 'ENTER'/'EXIT'/'DWELL' string on web and iOS, a raw GMS constant on Android. Modelling
 * it as one typed value gives the three dies a shared vocabulary. The string values are
 * exactly the public `GeofenceEvent.action` contract, so the emitted payload is
 * unchanged.
 */
export enum GeofenceTransition {
  ENTER = 'ENTER',
  EXIT = 'EXIT',
  DWELL = 'DWELL',
}
